src/soc/rockchip/rk3399.c

```diff
@@ -0,0 +1,538 @@
+	{"GPIO4_B6", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4B_IOMUX, 12}, {GPIO_SWPORTA_DDR, 14}, {GPIO_SWPORTA_DR, 14}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_B7", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4B_IOMUX, 14}, {GPIO_SWPORTA_DDR, 15}, {GPIO_SWPORTA_DR, 15}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C0", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 0}, {GPIO_SWPORTA_DDR, 16}, {GPIO_SWPORTA_DR, 16}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C1", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 2}, {GPIO_SWPORTA_DDR, 17}, {GPIO_SWPORTA_DR, 17}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C2", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 4}, {GPIO_SWPORTA_DDR, 18}, {GPIO_SWPORTA_DR, 18}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C3", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 6}, {GPIO_SWPORTA_DDR, 19}, {GPIO_SWPORTA_DR, 19}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C4", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 8}, {GPIO_SWPORTA_DDR, 20}, {GPIO_SWPORTA_DR, 20}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C5", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 10}, {GPIO_SWPORTA_DDR, 21}, {GPIO_SWPORTA_DR, 21}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C6", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 12}, {GPIO_SWPORTA_DDR, 22}, {GPIO_SWPORTA_DR, 22}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_C7", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4C_IOMUX, 14}, {GPIO_SWPORTA_DDR, 23}, {GPIO_SWPORTA_DR, 23}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D0", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 0}, {GPIO_SWPORTA_DDR, 24}, {GPIO_SWPORTA_DR, 24}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D1", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 2}, {GPIO_SWPORTA_DDR, 25}, {GPIO_SWPORTA_DR, 25}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D2", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 4}, {GPIO_SWPORTA_DDR, 26}, {GPIO_SWPORTA_DR, 26}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D3", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 6}, {GPIO_SWPORTA_DDR, 27}, {GPIO_SWPORTA_DR, 27}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D4", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 8}, {GPIO_SWPORTA_DDR, 28}, {GPIO_SWPORTA_DR, 28}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D5", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 10}, {GPIO_SWPORTA_DDR, 29}, {GPIO_SWPORTA_DR, 29}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D6", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 12}, {GPIO_SWPORTA_DDR, 30}, {GPIO_SWPORTA_DR, 30}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+	{"GPIO4_D7", 4, {CRU_CLKGATE_CON31, 5}, {GRF_GPIO4D_IOMUX, 14}, {GPIO_SWPORTA_DDR, 31}, {GPIO_SWPORTA_DR, 31}, FUNCTION_DIGITAL, PINMODE_NOT_SET, 0},
+};
+
+static int rk3399Setup(void) {
+	if((rk3399->fd = open("/dev/mem", O_RDWR | O_SYNC)) < 0) {
+		wiringXLog(LOG_ERR, "wiringX failed to open /dev/mem for raw memory access");
+		return -1;
+	}
+	for(int i = 0; i < GPIO_BANK_COUNT; i++) {
```

Comment:
Can you move the define for `i` above and not inside the for loop. This allows compilation in lower C versions.